sync.js

```diff
@@ -464,10 +471,13 @@ class Sync extends events.EventEmitter {
           // XXX: This is a hack to ensure sync streams always end eventually
           // Ideally, we'd open a sparse hypercore instead.
           heartbeat = setInterval(() => {
-            if (self.state.stale(peer)) {
-              connection.destroy(new Error('timed out due to missing data'))
+            var stale = self.state.stale(peer)
+            if (stale) {
+              var err = new Error('timed out due to missing data')
+              err.code = ERR_MISSING_DATA
```

Comment:
This is so the frontend can easily ignore or translate this error in the future. 

note to self: open issue on adding codes for all mapeo core erorrs here. (and/or does that issue exist already?)

Reply:
There's an `errors.js` on `master` for this, but, like so many things, it did not get backported.

Reply:
Got it!